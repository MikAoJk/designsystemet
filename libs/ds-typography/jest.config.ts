export default {
  displayName: 'ds-typography',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-typography',
};
