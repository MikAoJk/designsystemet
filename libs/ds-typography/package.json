{
  "name": "@skatteetaten/ds-typography",
  "version": "0.0.1",
  "groupId": "no.skatteetaten.aurora",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18"
  }
}
