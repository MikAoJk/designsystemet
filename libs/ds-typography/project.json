{
  "name": "ds-typography",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-typography/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-typography",
        "tsConfig": "libs/ds-typography/tsconfig.lib.json",
        "project": "libs/ds-typography/package.json",
        "entryFile": "libs/ds-typography/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-typography/rollup.config.js"],
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-typography/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/ds-typography/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-typography/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-typography/**/*.css",
          "libs/ds-typography/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-typography"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish {args.switches}",
        "cwd": "dist/libs/ds-typography"
      }
    }
  },
  "implicitDependencies": ["!ds-dev-config"]
}
