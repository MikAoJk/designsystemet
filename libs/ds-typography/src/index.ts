export * from './Blockquote/Blockquote';
export * from './Blockquote/Blockquote.types';
export * from './Paragraph/Paragraph';
export * from './Paragraph/Paragraph.types';
