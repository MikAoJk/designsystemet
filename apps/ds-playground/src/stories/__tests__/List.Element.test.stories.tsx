import { List } from '@skatteetaten/ds-typography';
import { expect } from '@storybook/jest';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { within } from '@storybook/testing-library';

export default {
  component: List.Element,
  title: 'Tester/List.Element',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: 'text',
    },
  },
} as ComponentMeta<typeof List.Element>;

const defaultArgs = {
  children: 'Kjenner du behovet til brukeren?',
};
const Template: ComponentStory<typeof List.Element> = (args) => (
  <ul data-test-block>
    <List.Element {...args} />
  </ul>
);

// Når ListElement har en ref, så får dom elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLLIElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ...WithRef.argTypes,
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const list = canvas.getByRole('listitem');
  await expect(list).toHaveAttribute('id', 'dummyIdForwardedFromRef');
};

// Når List.Element har en id, custom className, lang og dataTestid, så får elementet attributene id, class, lang og data-testid satt og custom stil vises
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes (FA2-5)';
WithAttributes.args = {
  ...defaultArgs,
  id: 'htmlid',
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
};
WithAttributes.argTypes = {
  ...WithAttributes.argTypes,
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const list = canvas.getByRole('listitem');
  await expect(list).toHaveAttribute('id', 'htmlid');
  await expect(list).toHaveClass('dummyClassname');
  await expect(list).toHaveAttribute('lang', 'nb');
  await expect(list).toHaveAttribute('data-testid', '123ID');
};
