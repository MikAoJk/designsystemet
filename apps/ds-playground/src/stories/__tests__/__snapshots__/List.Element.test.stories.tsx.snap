// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/List.Element WithCustomCss smoke-test 1`] = `
<ul data-test-block="true">
  <li class="ListElement_element__mEewo dummyClassname">
    Kjenner du behovet til brukeren?
  </li>
</ul>
`;

exports[`Tester/List.Element WithDataTestid smoke-test 1`] = `
<ul data-test-block="true">
  <li class="ListElement_element__mEewo "
      data-testid="ListElementID"
  >
    Kjenner du behovet til brukeren?
  </li>
</ul>
`;

exports[`Tester/List.Element WithId smoke-test 1`] = `
<ul data-test-block="true">
  <li id="ListElementId"
      class="ListElement_element__mEewo "
  >
    Kjenner du behovet til brukeren?
  </li>
</ul>
`;

exports[`Tester/List.Element WithLang smoke-test 1`] = `
<ul data-test-block="true">
  <li class="ListElement_element__mEewo "
      lang="nb"
  >
    Kjenner du behovet til brukeren?
  </li>
</ul>
`;

exports[`Tester/List.Element WithRef smoke-test 1`] = `
<ul data-test-block="true">
  <li class="ListElement_element__mEewo "
      id="dummyIdForwardedFromRef"
  >
    Kjenner du behovet til brukeren?
  </li>
</ul>
`;
